src/main/java/org/benetech/servicenet/adapter/DataAdapterFactory.java

```diff
@@ -18,15 +19,15 @@ public DataAdapterFactory(ApplicationContext context) {
     public Optional<SingleDataAdapter> getSingleDataAdapter(String providerName) {
         try {
             return Optional.of(context.getBean(providerName + DATA_ADAPTER_SUFFIX, SingleDataAdapter.class));
-        } catch (NoSuchBeanDefinitionException e) {
+        } catch (NoSuchBeanDefinitionException | BeanNotOfRequiredTypeException e) {
```

Comment:
Is this new exception something we handle silently like this? Seems like a configuration issue we would need to signal on

Reply:
BeanNotOfRequiredTypeException will be thrown when we'll ask for SingleDataAdapter, but the adapter for this partner is a type of MultipleData. In this case, we're simply returning an empty optional. Should it be changed?

Reply:
If this is something that should not happen in normal usage (and I assume it shouldn't?) we should raise an exception

Reply:
Ok, thanks